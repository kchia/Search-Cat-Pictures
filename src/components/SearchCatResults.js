import React from 'react';

function SearchCatResults(props) {
  const { images } = props;
  if (!images.length) {
    return <h2>No Images Found!</h2>;
  }
  return (
    <div className="gallery">
      {images.map(image => (
        <div id={image.id} key={image.id} className="gif">
          <img src={image.url} />
        </div>
      ))}
    </div>
  );
}

export default SearchCatResults;
