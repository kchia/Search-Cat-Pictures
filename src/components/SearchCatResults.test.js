import React from 'react';
import { shallow, mount } from 'enzyme';
// import SearchCatResults from './SearchCatResults';
import SearchCatResults from './SearchCatResults';
import { expect } from 'chai';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });
//we test if there are images
// Hou comment: Nice job writing tests for your component!
describe('SearchCatResults', () => {
  it('should render an image when passed an array of images prop', () => {
    const images = [
      {
        id: 'randomId',
        url: 'fakeurl.com'
      }
    ];
    const wrapper = mount(<SearchCatResults images={images} />);
    expect(wrapper.find('.gallery').children()).to.have.lengthOf(images.length);
    expect(
      wrapper
        .find('.gallery')
        .children()
        .at(0)
        .props().id
    ).to.equal('randomId');
    wrapper.unmount();
  });
  //we test it should render "no found images" if images not found
  it('should render No Images Found! if given an empty array', () => {
    const images = [];
    const wrapper = shallow(<SearchCatResults images={images} />);
    expect(wrapper.html()).to.equal('<h2>No Images Found!</h2>');
  });
});
