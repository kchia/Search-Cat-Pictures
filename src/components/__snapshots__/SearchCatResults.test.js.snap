// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchCatResults should render an image when passed an array of images prop 1`] = `ShallowWrapper {}`;
